### DISCUSSION TITLE
Mailserver up 15 minutes and already recieving spammer logins!

### DISCUSSION THREAD

**Discussion by Jarrodsz** [docker-mailserver/docker-mailserver#3311] [Q&A] [answered]
```
mailserver    | May  5 14:25:46 mail postfix/smtpd[3419]: NOQUEUE: reject: RCPT from unknown[95.214.27.183]: 554 5.7.1 <spameri@tiscali.it>: Relay access denied; from=<spameri@tiscali.it> to=<spameri@tiscali.it> proto=ESMTP helo=<WIN-CLJ1B0GQ6JP>
mailserver    | May  5 14:25:46 mail postfix/smtpd[3419]: disconnect from unknown[95.214.27.183] ehlo=1 mail=1 rcpt=0/1 rset=1 quit=1 commands=4/5
```

How do I prevent these things?

Would it be possible to auto load blacklists of spammers into docker mailserver so you can preventthose requests?

Else,
Would it be possible to generate a whitelist of ips or domains you connect from which are your own so it blocks all requests automatically before they "go true" the mailserver?

Perhaps i missed a part in the docs on "hardening" the config. But perhaps a usefull inprovement to docker mailserver would be to assist the user in setting up some defaults to prevent these kind of spammer login openrelay requests?

How about some spamhaus integration?

Im really new to the project just some ideas.

Would love to learn how to prevent any spam requests like these.

**Comment by polarathene** [comment, marked as answer]
Postfix log is just saying it received a connection from the spammer sending mail, who tried to relay through you instead of deliver mail to your domain. They're probably just a bot trying random IP to send mail through. Postfix denied the spammer.

If the spammer kept trying, you can have something like Fail2Ban notice by monitoring those logs and banning them. You're going to get plenty of these though with time. There is blacklists, but I believe the direction a maintainer of DMS is trying to migrate to is simplifying this for users with the rspamd feature that should replace many separate anti-spam features we have. I would suggest searching our issue history or looking through the docs ENV page, this isn't an area I'm well versed in about the project (_we do have spamhaus integration AFAIK, just required an ENV to enable for postscreen I think_).

You can whitelist too, but I am not sure if that is something we have simplified for users, it may require extra config. Generally though anything arriving on port 25 should be allowed to connect to deliver mail, most of the time it's probably a spammer trying to relay or bounce mail (backscatter), but if you only have a whitelist for that traffic, then you'd be preventing legitimate mail from third-parties.

**Reply by Jarrodsz** [reply]
Makes sense.
I will investigate more rspamd seems to fix a lot of these issues, and ir will be implemented in the near future from
Reading the issue tracker.

Id probably await that then and for now see if the spamassasins condiguration had it enabled.

Thank you
